src/module/module.py

```diff
@@ -19,23 +22,29 @@ def module_main():
 
     log.debug("Inputting data...")
 
-    try:
-        # YOUR CODE HERE
-        # ----------------------------------------------------------------
-
-        # input_data are data received by the module
-        input_data = None
-
-
-        # ----------------------------------------------------------------
-
-        # send data to the next module
-        send_error = send_data(input_data)
-
-        if send_error:
-            log.error(send_error)
-        else:
-            log.debug("Data sent sucessfully.")
-
-    except Exception as e:
-        log.error(f"Exception in the module business logic: {e}")
+    while True:
+        # incoming i2c byte from slave
+        if str(getenv('DATA_TYPE')) == "byte" :
+            byte_i2c = I2Cbus.read_byte_data(int(getenv('SLAVE_ADDR')), OFFSET)
+            byte_i2c_data = {"i2cData": byte_i2c }
+            print("I2C byte :  ",byte_i2c_data)
+            # send data to the next module
+            send_error = send_data(byte_i2c_data)
+
+            if send_error:
+                log.error(send_error)
+            else:
+                log.debug("Data sent sucessfully.")
+         # incoming i2c word from slave
+        if str(getenv('DATA_TYPE')) == "word" :
+            word_i2c = I2Cbus.read_word_data(int(getenv('SLAVE_ADDR')), OFFSET)
+            word_i2c_data = {"i2cData": word_i2c }
+            print("I2C word :  ",word_i2c_data)
+            # send data to the next module
+            send_error = send_data(word_i2c_data)
+
+            if send_error:
+                log.error(send_error)
+            else:
+                log.debug("Data sent sucessfully.")
+        sleep(int(getenv('PERIOD')))
```

Comment:
what if the slave doesn't send the data periodically? what happens if it sends a message while the module sleeps?

Reply:
in this case, the master will not read the data, Christos informs me to add that period even I don't agree , I see we need to make a delay only in the dev stage.

Reply:
after thinking about it I found that the slave requires that the master asks for data to send it including the period will not cause missing data.

Reply:
correct: slave is slave and has no willpower.  slave doesn't send, master reads from it.